examples/src/main/kotlin/example/ForwardingRefs.kt

```diff
@@ -0,0 +1,41 @@
+package example
+
+import org.w3c.dom.HTMLInputElement
+import react.*
+import react.dom.html.ReactHTML.input
+
+external interface Focusable {
+    fun focus()
+}
+
+val FancyInput = forwardRef<Props> { _, forwardedRef ->
+    val inputRef = useRef<HTMLInputElement>()
+
+    useImperativeHandle(forwardedRef, inputRef) {
+        object : Focusable {
+            override fun focus() {
+                inputRef.current?.focus()
+            }
+        }
+    }
+
+    input {
+        attrs {
+            ref = inputRef
+        }
+    }
+}
+
+val App = fc<Props> {
+    val fancyInputRef = useRef<Focusable>()
+
+    useEffectOnce {
+        fancyInputRef.current?.focus()
+    }
+
+    FancyInput {
+        attrs {
+            ref = fancyInputRef
```

Comment:
Request discussion about "forward ref component"

Reply:
All in all, I think we can merge this PR